src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -1451,15 +1501,20 @@ public async Task<string> GetStringValue(SessionId sessionId, int string_id, Can
             }
             return null;
         }
-        public async Task<int> GetArrayLength(SessionId sessionId, int object_id, CancellationToken token)
+        public async Task<ArrayDimensions> GetArrayLength(SessionId sessionId, int object_id, CancellationToken token)
         {
             var commandParams = new MemoryStream();
             var commandParamsWriter = new MonoBinaryWriter(commandParams);
             commandParamsWriter.Write(object_id);
             var retDebuggerCmdReader = await SendDebuggerAgentCommand<CmdArray>(sessionId, CmdArray.GetLength, commandParams, token);
             var length = retDebuggerCmdReader.ReadInt32();
-            length = retDebuggerCmdReader.ReadInt32();
-            return length;
+            var arrDimensions = new ArrayDimensions(length);
```

Comment:
should this be `var rank = new int[length]` then `new ArrayDimensions(rank)` and forgo the resizing?

Reply:
And populate `rank`, before `new ArrayDimensions`, so `TotalLength` can be precomputed too. And `ArrayDimensions` can be a `record`.

Reply:
done.